core/test/allocation.jl

```diff
@@ -22,7 +22,54 @@ using PreallocationTools: get_tmp
     @test F ≈ [0.0, 4.0, 0.0, 0.0, 0.0, 4.5]
 
     allocated = p.user.allocated
-    @test allocated[1] ≈ [4.0, 4.0]
-    @test allocated[2] ≈ [0.0, 0.0]
+    @test allocated[1] ≈ [0.0, 4.0]
+    @test allocated[2] ≈ [4.0, 0.0]
     @test allocated[3] ≈ [0.0, 0.0]
 end
+
+@testset "Simulation with allocation" begin
+    toml_path =
+        normpath(@__DIR__, "../../generated_testmodels/simple_subnetwork/ribasim.toml")
+    @test ispath(toml_path)
+    model = Ribasim.run(toml_path)
+    record = model.integrator.p.user.record
```

Comment:
Since this is a NamedTuple of vectors, this could be converted to a DataFrame directly in the tests. This could make the tests easier to understand.